File: services/processing-core/app/Infrastructure/Messaging/RabbitMq/RabbitMqConnectionFactory.php
<?php

namespace App\Infrastructure\Messaging\RabbitMq;

use PhpAmqpLib\Connection\AMQPStreamConnection;

/**
 * Fabrika za RabbitMQ konekciju u processing servisu.
 */
final class RabbitMqConnectionFactory
{
    /**
     * Kreira novu RabbitMQ konekciju na osnovu konfiguracije.
     */
    public function make(): AMQPStreamConnection
    {
        return new AMQPStreamConnection(
            (string) config('rabbitmq.host'),
            (int) config('rabbitmq.port'),
            (string) config('rabbitmq.user'),
            (string) config('rabbitmq.password'),
            (string) config('rabbitmq.vhost'),
        );
    }
}

File: services/processing-core/src/Infrastructure/Messaging/RabbitMq/RabbitMqConnectionFactory.cs
using Infrastructure.Config;
using RabbitMQ.Client;

namespace Infrastructure.Messaging.RabbitMq;

/// <summary>
/// Pravi RabbitMQ konekcije na osnovu konfiguracije servisa.
/// </summary>
public sealed class RabbitMqConnectionFactory
{
    private readonly RabbitMqConfig config;

    public RabbitMqConnectionFactory(RabbitMqConfig config)
    {
        this.config = config;
    }

    /// <summary>
    /// Otvara novu RabbitMQ konekciju.
    /// </summary>
    public IConnection CreateConnection()
    {
        var factory = new ConnectionFactory
        {
            HostName = config.Host,
            Port = config.Port,
            UserName = config.User,
            Password = config.Password,
            VirtualHost = config.VHost,
            DispatchConsumersAsync = false,
        };

        return factory.CreateConnection();
    }
}
